pkg/crds/crds.go

```diff
@@ -18,10 +18,185 @@ limitations under the License.
+	}
+}
+
+func (p *pathConfigs) addAnnotationPathConfig(config transformers.PathConfig) {
+	if *(p.annotationPathConfig.GroupVersionKind) == *config.GroupVersionKind {
+		p.annotationPathConfig.Path = append(p.labelPathConfig.Path, config.Path...)
+	} else {
+		p.annotationPathConfig = config
+	}
+}
+
+func (p *pathConfigs) addNamereferencePathConfig(config transformers.ReferencePathConfig) {
+	p.namereferencePathConfigs = transformers.MergeNameReferencePathConfigs(p.namereferencePathConfigs, config)
+}
+
+func (p *pathConfigs) addPrefixPathConfig(config transformers.PathConfig) {
+	if *(p.prefixPathConfig.GroupVersionKind) == *config.GroupVersionKind {
+		p.prefixPathConfig.Path = append(p.prefixPathConfig.Path, config.Path...)
+	} else {
+		p.prefixPathConfig = config
+	}
+}
+
 // RegisterCRDs parse CRD schemas from paths and update various pathConfigs
 func RegisterCRDs(loader loader.Loader, paths []string) error {
+	pathConfigs := []pathConfigs{}
+	for _, path := range paths {
+		pathConfig, err := registerCRD(loader, path)
+		if err != nil {
+			return err
+		}
+		pathConfigs = append(pathConfigs, pathConfig...)
+	}
+	addPathConfigs(pathConfigs)
+	return nil
+}
+
+// register CRD from one path
+func registerCRD(loader loader.Loader, path string) ([]pathConfigs, error) {
+	result := []pathConfigs{}
+
+	content, err := loader.Load(path)
+	if err != nil {
+		return result, err
+	}
+
+	var types map[string]common.OpenAPIDefinition
+	if content[0] == '{' {
+		err = json.Unmarshal(content, &types)
+	} else {
+		err = yaml.Unmarshal(content, &types)
+	}
+
+	crds := getCRDs(types)
+	for _, crd := range crds {
+		crdPathConfigs := pathConfigs{}
+		err = getCRDPathConfig(types, crd, crd, []string{}, &crdPathConfigs)
+		if err != nil {
+			return result, err
+		}
+		if !reflect.DeepEqual(crdPathConfigs, pathConfigs{}) {
+			result = append(result, crdPathConfigs)
+		}
+	}
+
+	return result, nil
+}
+
+// getCRDs get all CRD types
+func getCRDs(types map[string]common.OpenAPIDefinition) []string {
+	crds := []string{}
+
+	for typename, t := range types {
+		properties := t.Schema.SchemaProps.Properties
+		if _, ok := properties["kind"]; ok {
+			if _, ok := properties["apiVersion"]; ok {
+				if _, ok := properties["metadata"]; ok {
+					crds = append(crds, typename)
+				}
+			}
+		}
+	}
+	return crds
+}
+
+// getCRDPathConfig gets pathConfigs for one CRD recursively
+func getCRDPathConfig(types map[string]common.OpenAPIDefinition, atype string, crd string,
+	path []string, configs *pathConfigs) error {
+	if _, ok := types[crd]; !ok {
+		return nil
+	}
+
+	crdname := strings.Split(crd, ".")[len(strings.Split(crd, "."))-1]
+	for propname, property := range types[atype].Schema.SchemaProps.Properties {
+		_, annotate := property.Extensions.GetString(Annotation)
+		if annotate {
+			configs.addAnnotationPathConfig(
+				transformers.PathConfig{
+					CreateIfNotPresent: false,
+					GroupVersionKind:   &schema.GroupVersionKind{Kind: crdname},
```

Comment:
Only using the kind name can collide with another CRD.
If you have changed the signature of this method, you can use the passed-in `crd GroupVersionKind`.